yt_dlp/extractor/downdogapp.py

```diff
@@ -0,0 +1,191 @@
+        if login_result['type'] == 'WRONG_PASSWORD':
+            raise ExtractorError(
+                'Login failed (invalid username/password)', expected=True,
+            )
+        elif login_result['type'] == 'SUCCESS':
+            new_credential = login_result['cred']
+            self._set_cookie(hostname, 'credentials', new_credential)
+        else:  # unexpected error
+            raise ExtractorError('Login failed (unexpected)')
+
+    def _get_credential_key(self, hostname):
+        cookie = self._get_cookies(f'https://{hostname}')
+
+        if cookie:
+            return cookie['credentials'].coded_value
+        else:
+            return None
+
+    def _real_extract(self, url):
+        hostname = urllib.parse.urlparse(url).hostname
+        username, password = self._get_login_info(netrc_machine=hostname)
+        if username:
+            self._login(hostname, username, password)
+
+        sequence_id = self._match_id(url)
+
+        api_key = self._get_credential_key(hostname)
+        if not api_key:
+            # most likely not logged in
+            raise ExtractorError(
+                'Unable to get access token, did you pass cookies or log in?',
+                expected=True,
+            )
+        # These parameters seem to be irrelevant to what is returned,
+        #  but are require to get a proper return.
+        # Prepare the parameters for the POST data
+        _params = self._generate_device_description()
+        device = _params['deviceDescription']
+        os = _params['osVersion']
+        timezone = _params['timeZone']
+        unix_timestamp = _params['timestamp']
+        language = _params['languageOption']
+
+        # Build the POST data string, order matters
+        timings_data = (
+            f'sequenceId={sequence_id}&'
+            f'playlistSettings=%7B%2216%22%3A-1%7D&'
+            f'forOffline=false&'
+            f'deviceDescription={device}&'
+            f'osVersion={os}&'
+            f'timeZone={timezone}&'
+            f'locale=en-US&'
+            f'appType=ORIGINAL&'
+            f'timestamp={unix_timestamp}&'
+            f'cred={api_key}&'
+            f'languageOption={language}'
+        ).encode()
+
+        timings = self._request_webpage(
+            f'https://{hostname}/json/practice',
+            sequence_id,
+            data=timings_data,
+            headers={'Content-Type': 'application/x-www-form-urlencoded'},
+        )
+        timings_dict = self._parse_json(timings.read().decode('utf-8'), sequence_id)
+        try:
+            length_in_minutes = int(
+                timings_dict['sequence']['totalTime']['seconds'] // 60,
+            )
+            title = (
+                f'Down Dog {timings_dict["sequence"]["activityType"].capitalize()} -'
+                f' {timings_dict["sequence"]["trackerLabel"]} {length_in_minutes}min'
+            )
+        except KeyError:
+            length_in_minutes = 80  # to be safe
+            title = None
```

Comment:
instead of this try/catch, use `traverse_obj` to get the `seconds` value (converting to minutes is pointless), and use `join_nonempty` for the title